packages/catalog-client/src/CatalogClient.ts

```diff
@@ -346,7 +346,7 @@ export class CatalogClient implements CatalogApi {
     options?: CatalogRequestOptions,
   ): Promise<Location | undefined> {
     const all: { data: Location }[] = await this.requestRequired(
```

Comment:
I think it's worth fixing the `this.requestRequired` function so that we can actually make sure that the types that are returned are the correct ones. There could be other errors in this file that we haven't caught yet.

Reply:
We shouldn't need the const all: { data: Location }[], Think you can probably remove the <T=any> and let typescript work out the types dynamically, and it should throw some other errors in this file if there's other things that are awry.

Reply:
Thing is the generated client doesn't export `TypedResponse` so I'd maybe have to do some more fiddling then. Orrrrr I just ignore that and skip awaiting the inner one.

Reply:
Alright, updated the template to export the required type to get this right